cybershell/fingerprint.py

```
@@ -0,0 +1,649 @@
+                    if pattern.lower() in html_lower:
+                        if tech_name not in fingerprint.technologies:
+                            fingerprint.technologies.append(tech_name)
+                            fingerprint.confidence[f'tech_{tech_name}'] = tech_data['confidence']
+            
+            # Extract all script sources
+            scripts = soup.find_all('script', src=True)
+            fingerprint.raw_signals['scripts'] = [s['src'] for s in scripts]
+            
+            # Extract all link hrefs
+            links = soup.find_all('link', href=True)
+            fingerprint.raw_signals['stylesheets'] = [l['href'] for l in links if 'stylesheet' in l.get('rel', [])]
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)